docs/design/hub-and-spoke/switch-validate/switch-validate.md

```diff
@@ -5,9 +5,680 @@
+* groups[excess], meters[excess] and rule[excess] do not contain flow info.
+
+Response payload
+
+```json
+{
+  "as_expected": true,
+  "groups": {
+    "as_expected": true,
+    "excess": [
+      <v2GroupInfo> // without flow_id and flow_path
+    ],
+    "missing": [
+      <v2GroupInfo>
+    ],
+    "proper": [
+      <v2GroupInfo>
+    ],
+    "misconfigured": [
+      {
+        "id": "12123213",
+        "expected": {
+          <v2GroupInfo>
+        },
+        "discrepancies": {
+          <v2GroupInfo>
+        }
+      }
+    ]
+  },
+  "logical_ports": {
+    "error": "some_error",
+    "as_expected": true,
+    "excess": [
+      <v2LogicalPortInfo>
+    ],
+    "missing": [
+      <v2LogicalPortInfo>
+    ],
+    "proper": [
+      <v2LogicalPortInfo>
+    ],
+    "misconfigured": [
+      {
+        "id": "123123",
+        "expected": {
+          <v2LogicalPortInfo>
+        },
+        "discrepancies": {
+          <v2LogicalPortInfo>
+        }
+      }
+    ]
+  },
+  "meters": {
+    "as_expected": true,
+    "excess": [
+      <v2MeterInfo> // without flow_id, y_flow_id and flow_path
+    ],
+    "missing": [
+      <v2MeterInfo>
+    ],
+    "proper": [
+      <v2MeterInfo>
+    ],
+    "misconfigured": [
+      {
+        "id": "1232131",
+        "expected": {
+          <v2MeterInfo>
+        },
+        "discrepancies": {
+          <v2MeterInfo>
+        }
+      }
+    ]
+  },
+  "rules": {
+    "as_expected": true,
+    "excess": [
+      <v2RuleInfo> // without flow_id, y_flow_id and flow_path
+    ],
+    "missing": [
+      <v2RuleInfo>
+    ],
+    "proper": [
+      <v2RuleInfo>
+    ],
+    "misconfigured": [
+      {
+        "id": "asdsad",
+        "expected": {
+          <v2RuleInfo>
+        },
+        "discrepancies": {
+          <v2RuleInfo>
+        }
+      }
+    ]
+  }
+}
+```
+
+### Validate rules, meters, lags and groups installed on the switch with filters
+
+### Including specific info in response
+
+Request params
+
+To get only group info in response use query param `group`, other fields will be omitted.
+
+`GET https://{host}/v2/switches/{switch_id}/validate?include=group`
+
+Response payload
+
+```json
+{
+  "groups": {
+    "as_expected": true,
+    "excess": [
+        <v2GroupInfo> // without flow_id and flow_path
+      ],
+      "missing": [
+        <v2GroupInfo>
+      ],
+      "proper": [
+        <v2GroupInfo>
+      ],
+      "misconfigured": [
+        {
+          "id": "12123213",
+          "expected": {
+            <v2GroupInfo>
+          },
+          "discrepancies": {
+            <v2GroupInfo>
+          }
+        }
+      ]
+    }
+}
+```
+
+All other targets can be selected in same way i.e `include=meters` for meters, `include=rules` and
+so on.
+
+You can combine several flags in one query e.g. to get full info only about groups and meters use query
+params `group` and `meter`.
+
+`GET https://{host}/v2/switches/{switch_id}/validate?include=group|meter`
```

Comment:
Mapping between `include` argument values and specific validataion target still required. i.e. we are passing `?include=group` but in response we will get `{"groups": {}}`.

Why not to use response field's name as value for `include`/`exclude` arguments? I.e. groups|logical_ports|meters|rules/ In this case such mapping will not be required.

Reply:
will use multiply names